Engine/Results/BacktestingResultHandler.cs

```diff
@@ -42,81 +41,75 @@ public class BacktestingResultHandler : BaseResultsHandler, IResultHandler
         private DateTime _nextSample;
-        private IMessagingHandler _messagingHandler;
-        private ITransactionHandler _transactionHandler;
-        private ISetupHandler _setupHandler;
         private string _algorithmId;
         private int _projectId;
 
-        private const double Samples = 4000;
-        private const double MinimumSamplePeriod = 4;
-
         /// <summary>
         /// Packeting message queue to temporarily store packets and then pull for processing.
         /// </summary>
-        public ConcurrentQueue<Packet> Messages { get; set; } = new ConcurrentQueue<Packet>();
-
-        /// <summary>
-        /// Local object access to the algorithm for the underlying Debug and Error messaging.
-        /// </summary>
-        public IAlgorithm Algorithm { get; set; }
+        public ConcurrentQueue<Packet> Messages { get; set; }
 
         /// <summary>
         /// Charts collection for storing the master copy of user charting data.
         /// </summary>
-        public ConcurrentDictionary<string, Chart> Charts { get; set; } = new ConcurrentDictionary<string, Chart>();
+        public ConcurrentDictionary<string, Chart> Charts { get; set; }
 
         /// <summary>
-        /// Boolean flag indicating the result hander thread is completely finished and ready to dispose.
+        /// Boolean flag indicating the result handler thread is completely finished and ready to dispose.
         /// </summary>
-        public bool IsActive { get; private set; } = true;
+        public bool IsActive { get; private set; }
 
         /// <summary>
         /// Sampling period for timespans between resamples of the charting equity.
         /// </summary>
         /// <remarks>Specifically critical for backtesting since with such long timeframes the sampled data can get extreme.</remarks>
-        public TimeSpan ResamplePeriod { get; private set; } = TimeSpan.FromMinutes(4);
+        public TimeSpan ResamplePeriod { get; private set; }
 
         /// <summary>
         /// How frequently the backtests push messages to the browser.
         /// </summary>
         /// <remarks>Update frequency of notification packets</remarks>
-        public TimeSpan NotificationPeriod { get; } = TimeSpan.FromSeconds(2);
+        public TimeSpan NotificationPeriod { get; }
 
         /// <summary>
         /// A dictionary containing summary statistics
         /// </summary>
         public Dictionary<string, string> FinalStatistics { get; private set; }
 
         /// <summary>
-        /// Default initializer for
+        /// Creates a new instance
         /// </summary>
         public BacktestingResultHandler()
         {
+            Messages = new ConcurrentQueue<Packet>();
+            Charts = new ConcurrentDictionary<string, Chart>();
+            IsActive = true;
+            ResamplePeriod = TimeSpan.FromMinutes(4);
+            NotificationPeriod = TimeSpan.FromSeconds(2);
```

Comment:
I like! Move all initialization to the same place, makes it easier to grock initial state 👍